src/Compilers/CSharp/Test/Syntax/Parsing/ParsingErrorRecoveryTests.cs

```diff
@@ -7056,6 +7078,26 @@ class c
             Assert.False(ns.CloseBraceToken.IsMissing);
         }
 
+
+        [WorkItem(947819, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/947819")]
+        [Fact]
+        public void MissingOpenBraceForClassSingleLineNamespace()
+        {
+            var source = @"namespace n;
+
+class c
+";
+            var root = SyntaxFactory.ParseSyntaxTree(source).GetRoot();
+
+            Assert.Equal(source, root.ToFullString());
+            // Verify incomplete class decls don't eat tokens of surrounding nodes
+            var classDecl = root.DescendantNodes().OfType<ClassDeclarationSyntax>().Single();
+            Assert.False(classDecl.Identifier.IsMissing);
+            Assert.True(classDecl.OpenBraceToken.IsMissing);
+            Assert.True(classDecl.CloseBraceToken.IsMissing);
+            var ns = root.DescendantNodes().OfType<SingleLineNamespaceDeclarationSyntax>().Single();
```

Comment:
Did you intend to check something on `ns`?

Reply:
@RikkiGibson to address

Reply:
It's not that meaningful but I'll add an assert that the semicolon is present.